### PR TITLE
Added setter check

### PR DESCRIPTION
**Pull request by gaurav1086**
Check setter before setter->getDeclContext().

Resolves SR-NNNN.

### CODE REVIEW

File: lib/ClangImporter/ImportDecl.cpp [hunk 1]
```diff
@@ -6759,7 +6759,7 @@ SwiftDeclConverter::importSubscript(Decl *decl,
   // - There is no setter.
   bool associateWithSetter = !getterAndSetterInSameType && setter == decl;
   DeclContext *dc =
-      associateWithSetter ? setter->getDeclContext() : getter->getDeclContext();
+      associateWithSetter ? setter && setter->getDeclContext() : getter->getDeclContext();
```
Comment (theblixguy) [inline lib/ClangImporter/ImportDecl.cpp]: If `setter` was null then `associateWithSetter` would be `false`

Comment (gaurav1086) [reply lib/ClangImporter/ImportDecl.cpp]: @theblixguy thanks for the review. Made some modifications.

File: lib/ClangImporter/ImportDecl.cpp [hunk 2]
```diff
@@ -6757,9 +6757,8 @@ SwiftDeclConverter::importSubscript(Decl *decl,
   // - A getter is redeclared in a subclass, but not the setter.
   // - The getter and setter are part of the same type.
   // - There is no setter.
-  bool associateWithSetter = !getterAndSetterInSameType && setter == decl;
   DeclContext *dc =
-      associateWithSetter ? setter->getDeclContext() : getter->getDeclContext();
+      !getterAndSetterInSameType && setter == decl ? setter->getDeclContext() : getter->getDeclContext();
```
Comment (theblixguy) [inline lib/ClangImporter/ImportDecl.cpp]: Sorry, what I meant before was that `setter` would never be null if `associateWithSetter` was false, which meant `setter->getDeclContext()` is perfectly fine to do. The null check is not needed.

I don't think this improves anything as it's just inlining the `associateWithSetter` computation.

Comment (gaurav1086) [reply lib/ClangImporter/ImportDecl.cpp]: Closing this.

**Pull request closed without merging**